node/core/approval-voting/src/old_tests.rs

```diff
@@ -0,0 +1,1964 @@
+		tranche,
+		validator_index: ValidatorIndex(0),
+		triggered: false,
+	}.into());
+
+	f(&mut candidate_entry);
+
+	overlay_txn(db, |overlay_db| overlay_db.write_candidate_entry(candidate_entry.clone()));
+}
+
+#[test]
+fn rejects_bad_assignment() {
+	let mut db = make_db();
+	let block_hash = Hash::repeat_byte(0x01);
+	let candidate_hash = CandidateReceipt::<Hash>::default().hash();
+	let assignment_good = IndirectAssignmentCert {
+		block_hash,
+		validator: ValidatorIndex(0),
+		cert: garbage_assignment_cert(
+			AssignmentCertKind::RelayVRFModulo {
+				sample: 0,
+			},
+		),
+	};
+	let mut state = some_state(StateConfig {
+		candidate_hash: Some(candidate_hash),
+		..Default::default()
+	}, &mut db);
+	let candidate_index = 0;
+
+	let mut overlay_db = OverlayedBackend::new(&db);
+	let res = check_and_import_assignment(
+		&mut state,
+		&mut overlay_db,
+		assignment_good.clone(),
+		candidate_index,
+	).unwrap();
+	assert_eq!(res.0, AssignmentCheckResult::Accepted);
+	// Check that the assignment's been imported.
+	assert_eq!(res.1.len(), 1);
+
+	let write_ops = overlay_db.into_write_ops().collect::<Vec<BackendWriteOp>>();
+	assert_eq!(write_ops.len(), 1);
+	assert_matches!(write_ops.get(0).unwrap(), BackendWriteOp::WriteCandidateEntry(..));
+	db.write(write_ops).unwrap();
+
+	// unknown hash
+	let unknown_hash = Hash::repeat_byte(0x02);
+	let assignment = IndirectAssignmentCert {
+		block_hash: unknown_hash,
+		validator: ValidatorIndex(0),
+		cert: garbage_assignment_cert(
+			AssignmentCertKind::RelayVRFModulo {
+				sample: 0,
+			},
+		),
+	};
+
+	let mut overlay_db = OverlayedBackend::new(&db);
+	let res = check_and_import_assignment(
+		&mut state,
+		&mut overlay_db,
+		assignment,
+		candidate_index,
+	).unwrap();
+	assert_eq!(res.0, AssignmentCheckResult::Bad(AssignmentCheckError::UnknownBlock(unknown_hash)));
+	assert_eq!(overlay_db.into_write_ops().count(), 0);
+
+	let mut state = State {
+		assignment_criteria: Box::new(MockAssignmentCriteria::check_only(|| {
+			Err(criteria::InvalidAssignment)
+		})),
+		..some_state(StateConfig {
+			candidate_hash: Some(candidate_hash),
+			..Default::default()
+		}, &mut db)
+	};
+
+	// same assignment, but this time rejected
+	let mut overlay_db = OverlayedBackend::new(&db);
+	let res = check_and_import_assignment(
+		&mut state,
+		&mut overlay_db,
+		assignment_good,
+		candidate_index,
+	).unwrap();
+	assert_eq!(res.0, AssignmentCheckResult::Bad(AssignmentCheckError::InvalidCert(ValidatorIndex(0))));
+	assert_eq!(overlay_db.into_write_ops().count(), 0);
+}
+
+#[test]
+fn rejects_assignment_in_future() {
+	let mut db = make_db();
+	let block_hash = Hash::repeat_byte(0x01);
+	let candidate_index = 0;
+	let candidate_hash = CandidateReceipt::<Hash>::default().hash();
+	let assignment = IndirectAssignmentCert {
+		block_hash,
+		validator: ValidatorIndex(0),
+		cert: garbage_assignment_cert(
+			AssignmentCertKind::RelayVRFModulo {
+				sample: 0,
+			},
+		),
+	};
+
+	let tick = 9;
```

Comment:
In this test are quite a few :mage_man: magic numbers, it would be nice to add a rationale why these are as they are.